### ISSUE TITLE
Extra argument

### ISSUE THREAD

**Issue by xogeny** [jsonata-js/jsonata#128]
In the following code:

```
if(isFunction(lhs)) {
    // this is function chaining (func1 ~> func2)
    // λ($f, $g) { λ($x){ $g($f($x)) } }
    result = yield * apply(chain, [lhs, func], environment, null);
} else {
    result = yield * apply(func, [lhs], environment, null);
}
```

`apply` is called with four arguments.  But it only takes three.  It doesn't hurt anything and I've fixed it in some code I'm working on.  But I just wanted to point it out so I could reference this issue in a future PR.

**Comment by andrew-coleman** [comment]
Thanks for that.  `apply` used to take 4 args, but I removed the need for the last arg in a previous release.  I'd clearly missed this one.  I guess TypeScript helps spot these issues :)

**Comment by mattbaileyuk** [comment]
Fixed in #129
